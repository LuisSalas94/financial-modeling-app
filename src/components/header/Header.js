import React from 'react';
// react-icons
import { IoIosArrowBack } from 'react-icons/io';
import { BsMicFill } from 'react-icons/bs';
import { AiTwotoneSetting } from 'react-icons/ai';
// router
import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <nav className="nav">
      <ul className="nav-ul">
        <li className="back">
          <Link to="/">
            <IoIosArrowBack />
          </Link>
        </li>
        <li className="logo">
          <Link to="/" className="logo">
            <h1 className="title">Financial Modeling Prep</h1>
          </Link>
        </li>
        <li className="icons">
          <button type="button" className="btn">
            <BsMicFill />
          </button>
          <button type="button" className="btn">
            <AiTwotoneSetting />
          </button>
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
