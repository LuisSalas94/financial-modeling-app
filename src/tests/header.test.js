import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Header from '../components/header/Header';

const MockedHeader = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  </Provider>
);

describe('Header component', () => {
  it('renders succesfully', () => {
    render(<MockedHeader />);
    const listedItem = screen.getByText('Financial Modeling Prep');
    expect(listedItem).toMatchSnapshot();
  });
});
