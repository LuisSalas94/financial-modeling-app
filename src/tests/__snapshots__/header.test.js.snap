// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders succesfully 1`] = `
<h1
  class="title"
>
  Financial Modeling Prep
</h1>
`;
